### PR TITLE
Add resource google_endpoints_service_consumers_iam

### PR DESCRIPTION
**Pull request by ScottSuarez**
closes https://github.com/hashicorp/terraform-provider-google/issues/9759

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_endpoints_service_consumers_iam_binding`
`google_endpoints_service_consumers_iam_member`
`google_endpoints_service_consumers_iam_policy`
```

### CODE REVIEW

File: mmv1/products/servicemanagement/api.yaml [hunk 1]
```diff
@@ -36,3 +36,18 @@ objects:
...
+    description: |
+      A service that is managed by Google Service Management
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'serviceName'
+        description: The name of the service.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'consumerProject'
+        description: Service consumer level permissions require that the request to the Service Control API must contain at least one valid service consumer project ID.
```
Comment (slevenick) [inline mmv1/products/servicemanagement/api.yaml:52]: Can we improve the description here? Is this the project that IAM is granting permission to or for, or how would a user decide what the consumer project is?

**Review by slevenick — APPROVED** [review]

File: mmv1/products/servicemanagement/terraform.yaml [hunk 2]
```diff
@@ -33,6 +33,30 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
+      example_config_body: 'templates/terraform/iam/example_config_body/service_management_consumer.tf.erb'
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "endpoints_iam"
+        primary_resource_id: "endpoints_service"
+        primary_resource_name: "fmt.Sprintf(\"endpoint%s.endpoints.%s.cloud.goog\", context[\"random_suffix\"], context[\"project_name\"]), context[\"project_name\"]"
+        vars:
+          name: "endpoint"
+        test_env_vars:
+          project_name: :PROJECT_NAME
```
[the remark this answers is not in the document]
Comment (ScottSuarez) [reply mmv1/products/servicemanagement/terraform.yaml:56]: correlates to `endpoints_iam` testcase 

I am reusing this testcase with injected code for the body of the new resource's block
https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/examples/endpoints_iam.tf.erb

File: mmv1/products/servicemanagement/api.yaml (hunk 1 above)
Comment (ScottSuarez) [reply mmv1/products/servicemanagement/api.yaml:52]: Starting discussion with the team to try and understand this field with more clarity.

File: mmv1/products/servicemanagement/terraform.yaml [hunk 3]
```diff
@@ -33,6 +33,30 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
+    id_format: 'services/{{service_name}}/consumers/{{consumer_project}}'
+    import_format: ["services/{{service_name}}/consumers/{{consumer_project}}", "{{consumer_project}}"]
+    exclude_resource: true
+    iam_policy: !ruby/object:Api::Resource::IamPolicy
+      method_name_separator: ':'
+      parent_resource_type: 'google_endpoints_service'
+      parent_resource_attribute: 'consumer_project'
+      allowed_iam_role: 'roles/servicemanagement.serviceController'
+      fetch_iam_policy_verb: :POST
+      exclude: false
```
[the remark this answers is not in the document]
Comment (ScottSuarez) [reply mmv1/products/servicemanagement/terraform.yaml]: done

**Pull request merged**